### ISSUE TITLE
Coordinate system for armies & army controllers?

### ISSUE THREAD

**Issue by PatrikLundell** [DFHack/df-structures#393]
Someone has labeled fields in the army and army_controller structs as coordinates, and so presumably had an idea of what those coordinates were. I've tried to make sense of them, but the numbers are too large both for the world tile and MLT scales. However, they may match a scale of 3 * 16 * world tile (i.e. number of map blocks), which is something I haven't encountered anywhere else.

Does anyone know what the previous thoughts about the coordinate system used in this context were?

**Comment by lethosor** [comment]
Looks like these were last changed in [2014](https://github.com/DFHack/df-structures/blame/0.47.04-beta1/df.military.xml#L353) and [2015](https://github.com/DFHack/df-structures/blame/0.47.04-beta1/df.military.xml#L626), if these are what you're referring to. Map block indices could be plausible - I think armies might be displayed at the block level in the adventure mode map.

**Comment by PatrikLundell** [comment]
Thanks
I ended up making the map block index assumption in my latest update (which you just reviewed), but it's good to get confirmation that it's reasonable. I did manage to match the coordinates with my fortress as well as the return to two attacker sites.
I also made the assumption that crime coordinates were in the in-game tile (2 * 2 m) coordinate system, as crimes happening in the fortress would end up there if those were the coordinates used.
